File: Backend/src/main/java/support/backend/Model/Panne.java
package support.backend.Model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Panne {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int idPanne;
    private String description;
    private LocalDate dateSignalement;
    private EtatPanne etatPanne;

    @OneToMany(mappedBy = "panne")
    private List<HistoriquePanne> historiquePannes;
}

File: Frontend/Frontend/src/app/interfaces/Panne.ts
export enum EtatPanne {

    'SIGNALEE',
    'EN_COURS',
    'RESOLUE',
    'NON_REPARABLE'

}


export interface Panne {
    idPanne?: number;
    description : string;
    etatPanne : EtatPanne;
}
